pkg/vetter/util/istio-inject.go

```diff
@@ -0,0 +1,613 @@
+		"excludeInboundPort":  excludeInboundPort,
+		"includeInboundPorts": includeInboundPorts,
+		"kubevirtInterfaces":  kubevirtInterfaces,
+		"applicationPorts":    applicationPorts,
+		"annotation":          annotation,
+		"valueOrDefault":      valueOrDefault,
+		"toJSON":              toJSON,
+		"toJson":              toJSON, // Used by, e.g. Istio 1.0.5 template sidecar-injector-configmap.yaml
+		"fromJSON":            fromJSON,
+		"toYaml":              toYaml,
+		"indent":              indent,
+		"directory":           directory,
+	}
+
+	var tmpl bytes.Buffer
+	temp := template.New("inject").Delims(sidecarTemplateDelimBegin, sidecarTemplateDelimEnd)
+	t, err := temp.Funcs(funcMap).Parse(sidecarTemplate)
+	if err != nil {
+		glog.Infof("Failed to parse template: %v %v\n", err, sidecarTemplate)
+		return nil, "", err
+	}
+	if err := t.Execute(&tmpl, &data); err != nil {
+		glog.Infof("Invalid template: %v %v\n", err, sidecarTemplate)
+		return nil, "", err
+	}
+
+	var sic SidecarInjectionSpec
+	if err := yaml.Unmarshal(tmpl.Bytes(), &sic); err != nil {
+		glog.Warningf("Failed to unmarshall template %v %s", err, tmpl.String())
+		return nil, "", err
+	}
+
+	// set sidecar --concurrency
+	// ASPENMESH - Comment out as not needed
+	// applyConcurrency(sic.Containers)
+
+	status := &SidecarInjectionStatus{Version: version}
+	for _, c := range sic.InitContainers {
+		status.InitContainers = append(status.InitContainers, c.Name)
+	}
+	for _, c := range sic.Containers {
+		status.Containers = append(status.Containers, c.Name)
+	}
+	for _, c := range sic.Volumes {
+		status.Volumes = append(status.Volumes, c.Name)
+	}
+	for _, c := range sic.ImagePullSecrets {
+		status.ImagePullSecrets = append(status.ImagePullSecrets, c.Name)
+	}
+	statusAnnotationValue, err := json.Marshal(status)
+	if err != nil {
+		return nil, "", fmt.Errorf("error encoded injection status: %v", err)
+	}
+	return &sic, string(statusAnnotationValue), nil
+}
+
+func getPortsForContainer(container corev1.Container) []string {
+	parts := make([]string, 0)
+	for _, p := range container.Ports {
+		parts = append(parts, strconv.Itoa(int(p.ContainerPort)))
+	}
+	return parts
+}
+
+func getContainerPorts(containers []corev1.Container, shouldIncludePorts func(corev1.Container) bool) string {
+	parts := make([]string, 0)
+	for _, c := range containers {
+		if shouldIncludePorts(c) {
+			parts = append(parts, getPortsForContainer(c)...)
+		}
+	}
+
+	return strings.Join(parts, ",")
+}
+
+func applicationPorts(containers []corev1.Container) string {
+	return getContainerPorts(containers, func(c corev1.Container) bool {
+		return c.Name != ProxyContainerName
+	})
+}
+
+func includeInboundPorts(containers []corev1.Container) string {
+	// Include the ports from all containers in the deployment.
+	return getContainerPorts(containers, func(corev1.Container) bool { return true })
+}
+
+func kubevirtInterfaces(s string) string {
+	return s
+}
+
+func toJSON(m map[string]string) string {
```

Comment:
Double check that both these and `ApplyJSON` are needed.

Reply:
They are fairly different (arguments, and parsing methods) so let's leave them both for now...